pkg/security/middlewares/authn.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package middlewares
+
+import (
+	"github.com/Peripli/service-manager/pkg/security"
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+// NewAuthnMiddleware returns web.Filter which uses the given security.Authenticator
+// to authenticate the request. FilterMatchers should be extended to cover the desired
+// endpoints.
+func NewAuthnMiddleware(filterName string, authenticator security.Authenticator) web.Filter {
```

Comment:
why return a filter when this is a middleware? to reuse the empty `FilterMatchers`? Better remove the FilterMatchers method

Reply:
BaseAuthnFilter?